### ISSUE TITLE
Could not evaluate: couldn't find login name -- expanding `~'

### ISSUE THREAD

**Issue by maxadamo** [sensu/sensu-puppet#1072]
## Description of problem

- if puppet is run unattended (as per defined schedule) it throws this error: 
```
Could not evaluate: couldn't find login name -- expanding `~'
```
- If puppet is run from a terminal, it works properly. 
- The error is only shown in the puppetserver.log, or in puppetDB (I have found it in the puppetboard, reporting tool).

## Command used and debugging output

The problem seems to be in the file: `lib/puppet/provider/sensuctl.rb`, at line 10 
```
home = File.expand_path('~')
```
The tilde is not expanded, and variable interpolation fails

## Platform and version information

- Your OS: Linux CentOS 7
- Your Ruby version: Sensu-Go
- Your version of Puppet: 6.2
- Your version of Sensu: 5.2.1
- Your version of this module: latest, branch sensu2

**Comment by maxadamo** [comment]
I could add in the file: 
```
require 'etc'
```
then replace line 10 with:
```
home = Etc.getpwnam('blahblah').dir
```
the user `blahblah` running puppet is root. So far so good, but I see either this:
```
/var/lib/sensu/.config/sensu/sensuctl/
```
and this:
```
/root/.config/sensu/sensuctl/
```

:question:

**Comment by ghoneycutt** [comment]
When you run puppet unattended, is that as the root user?

How can we reproduce this?

**Comment by maxadamo** [comment]
@ghoneycutt yes, it is the root user. 
The best way to reproduce it, is to set something like this in you puppet.conf: 
```
runinterval = 60
```
then check the logs of your puppetserver: `/var/log/puppetlabs/puppetserver/puppetserver.log`

I think you'll see the error as soon as puppet is triggered.

**Comment by maxadamo** [comment]
@ghoneycutt regression bug? It's happening again.

**Comment by treydock** [comment]
@maxadamo What's the version of this module you're on and what is the error exactly? Could you provide top few lines with '--trace' flag?

**Comment by maxadamo** [comment]
@treydock I upgraded from 3.14.0 to 4.5.1
To circumvent the issue I have created the file: `/etc/systemd/system/puppet.service.d/env.conf`
with the following content: 
```
[Service]
Environment="HOME=/root"
```

This time the issue is happening in the file: `manifests/cli.pp`
Line: 89

**Comment by maxadamo** [comment]
@treydock from my puppetboard:
![immagine](https://user-images.githubusercontent.com/1710103/75609838-22373500-5b0c-11ea-81ef-6aca6442842c.png)

**Comment by treydock** [comment]
I found the problem, we switched to `Dir.home` in Ruby but it does the same tilde expansion I think: https://bugs.ruby-lang.org/issues/12695.

A fix is in #1226 , let us know if that works for you.

**Comment by maxadamo** [comment]
well done @treydock 
it works
